### PR TITLE
Remove references to safe-string in the stdlib

### PR DESCRIPTION
**Pull request by hhugo**

### CODE REVIEW

**Comment by hhugo** [comment]
I wouldn't know what to write in a changelog entry.

File: stdlib/lexing.ml [hunk 1]
```diff
@@ -163,10 +163,11 @@ let from_channel ?with_positions ic =
   from_function ?with_positions (fun buf n -> input ic buf 0 n)
 
 let from_string ?(with_positions = true) s =
+  (* [lex_buffer] is exported in the mli, one could mutate it outside this module. *)
```
Comment (gasche) [inline stdlib/lexing.ml]: I find the comment a bit unclear: it explains why we need to make a copy, but without saying explicitly that we are (intentionally) making a copy.

Comment (hhugo) [reply stdlib/lexing.ml]: I've changed it a bit

**Review by gasche — APPROVED** [review]
Thanks!

**Comment by hhugo** [comment]
> Thanks!

Can you squash during the merge ?

**Comment by gasche** [comment]
check-typo says that you need to `make depend`. (I would recommend doing that and then squashing on your end.)

**Comment by hhugo** [comment]
fixed and squashed

**Comment by Octachron** [comment]
For the changelog entry, I would be partial to entry inside the documentation section of 5.3.

**Comment by Octachron** [comment]
Cherry-picked to 5.3 as 4d840b4aaee .

**Pull request merged**